R/utils-plumber.R

```diff
@@ -222,3 +222,60 @@ extract_endpoint <- function(path) {
   # stringr::str_extract(path, pattern = "([^/]+$)")
   sub(".*[/]", "", path)
 }
+
+
+#' Return the version of the data
+#'
+#' @param version Data version. Defaults to most recent version. See api/v1/versions
+#' @param release_version date when the data was published in YYYYMMDD format
+#' @param ppp_version ppp year to be used
+#' @param identity One of "PROD" (production), "INT" (internal) and "TEST"
+#' @param versions_available character vector of all the versions available
+#'
+#' @return character
+#'
+#' @export
+#'
+return_correct_version <- function(version = NULL,
+                                   release_version = NULL,
+                                   ppp_version = NULL,
+                                   identity = 'PROD',
+                                   versions_available) {
+  # If version is passed return it directly.
+  if (!is.null(version)) return(version)
+
+  if (!is.null(release_version) & !is.null(ppp_version) & !is.null(identity)) {
+    selected_version <- grep(sprintf('^%s_%s_\\d{2}_\\d{2}_%s$', release_version, ppp_version, identity), versions_available, value = TRUE)
+  } else if (!is.null(release_version) & !is.null(ppp_version)) {
+    # This probably would never be executed since identity would never be NULL.
+    selected_version <- grep(sprintf('^%s_%s_\\d{2}_\\d{2}_[A-Z]+$', release_version, ppp_version), versions_available, value = TRUE)
+  } else if (!is.null(release_version) & !is.null(identity)) {
+    selected_version <- grep(sprintf('^%s_\\d{4}_\\d{2}_\\d{2}_%s$', release_version, identity), versions_available, value = TRUE)
+  } else if (!is.null(ppp_version) & !is.null(identity)) {
+    selected_version <- grep(sprintf('\\d{6}_%s_\\d{2}_\\d{2}_%s$', ppp_version, identity), versions_available, value = TRUE)
+  }
```

Comment:
Would it make sense to create sub-functions that we can unit test here? I'm always a bit nervous when using `regex`...

Reply:
yes, that makes sense. I have done so and committed the change. However, I was hoping to find a solution which did not involve so many `if...else` statements but unable to come up with that logic. So we will keep this for now.